charms/kfp-persistence/src/charm.py

```diff
@@ -9,191 +9,71 @@
-
-        Raises:
-            CheckFailed(..., Blocked) when no relation established (unless default_return set)
-            CheckFailed(..., Blocked) if interface is not using SDI
-            CheckFailed(..., Blocked) if data in interface fails schema check
-            CheckFailed(..., Waiting) if we have a relation established but no data passed
-
-        Params:
-            interfaces:
-
-        Returns:
-              (dict) interface data
+
+        self.charm_executor.install(self)
+
+    def get_all_status(self):
+        """Convenience function for getting a list of all statuses for this charm's executor.
+
+        This is more for debugging.  Having an action would be useful to summarise this too on a
+        running charm.
         """
-        # If nothing is related to this relation, return a default value or raise an error
-        if relation_name not in interfaces or interfaces[relation_name] is None:
-            if default_return is not None:
-                return default_return
-            else:
-                raise CheckFailedError(
-                    f"Missing required relation for {relation_name}", BlockedStatus
-                )
-
-        relations = interfaces[relation_name]
-        if not isinstance(relations, SerializedDataInterface):
-            raise CheckFailedError(
-                f"Unexpected error with {relation_name} relation data - data not as expected",
-                BlockedStatus,
-            )
-
-        # Get and validate data from the relation
-        try:
-            # relations is a dict of {(ops.model.Relation, ops.model.Application): data}
-            unpacked_relation_data = relations.get_data()
-        except ValidationError as val_error:
-            # Validation in .get_data() ensures if data is populated, it matches the schema and is
-            # not incomplete
-            self.log.exception(val_error)
-            raise CheckFailedError(
-                f"Found incomplete/incorrect relation data for {relation_name}.  See logs",
-                BlockedStatus,
-            )
-
-        # Check if we have an established relation with no data exchanged
-        if len(unpacked_relation_data) == 0:
-            raise CheckFailedError(f"Waiting for {relation_name} relation data", WaitingStatus)
-
-        # Unpack data (we care only about the first element)
-        data_dict = list(unpacked_relation_data.values())[0]
-
-        # Catch if empty data dict is received (JSONSchema ValidationError above does not raise
-        # when this happens)
-        # Remove once addressed in:
-        # https://github.com/canonical/serialized-data-interface/issues/28
-        if len(data_dict) == 0:
-            raise CheckFailedError(
-                f"Found incomplete/incorrect relation data for {relation_name}.",
-                BlockedStatus,
-            )
-
-        return data_dict
+        return self.charm_executor.component_graph.status_prioritiser.all()
```

Comment:
Atm, this is unused afaict.  If it stays unused, we can remove it, but maybe it is useful somewhere.  I think it came from an action in one of the demo charms maybe?

Reply:
Removed.